src/Aspire.Dashboard/Components/Controls/ResourceDetails.razor

```diff
@@ -117,64 +117,61 @@
-                        </AspireTemplateColumn>
-                        <AspireTemplateColumn Title="@ControlStringsLoc[nameof(ControlsStrings.DetailsColumnHeader)]" Class="detailsColumn">
-                            <GridValue ValueDescription="@ControlStringsLoc[nameof(ControlsStrings.DetailsColumnHeader)]"
-                                       Value="@context.DisplayedDescription"
-                                       EnableHighlighting="@(!string.IsNullOrEmpty(_filter))"
-                                       HighlightText="@_filter"
-                                       TextVisualizerTitle="@context.Name">
-                                <ContentBeforeValue>
-                                    <ExceptionDetails ExceptionText="@context.ExceptionText" />
-                                </ContentBeforeValue>
-                            </GridValue>
-                        </AspireTemplateColumn>
-                    </FluentDataGrid>
-                }
+
+                <FluentDataGrid ResizeLabel="@AspireFluentDataGridHeaderCell.GetResizeLabel(ControlStringsLoc)"
+                                ResizeType="DataGridResizeType.Discrete"
+                                Items="@FilteredHealthReports"
+                                ItemKey="r => r.Name"
+                                ResizableColumns="true"
+                                Style="width:100%"
+                                GenerateHeader="GenerateHeaderOption.Sticky"
+                                GridTemplateColumns="1fr 1fr 1.5fr"
+                                ShowHover="true">
+                    <AspireTemplateColumn Title="@ControlStringsLoc[nameof(ControlsStrings.NameColumnHeader)]" Class="nameColumn">
+                        <GridValue ValueDescription="@ControlStringsLoc[nameof(ControlsStrings.NameColumnHeader)]"
+                                   Value="@context.Name"
+                                   EnableHighlighting="@(!string.IsNullOrEmpty(_filter))"
+                                   HighlightText="@_filter" />
+                    </AspireTemplateColumn>
+                    <AspireTemplateColumn Title="@ControlStringsLoc[nameof(ControlsStrings.StateColumnHeader)]" Class="stateColumn">
+                        <GridValue ValueDescription="@ControlStringsLoc[nameof(ControlsStrings.StateColumnHeader)]"
+                                   Value="@(context.HealthStatus?.Humanize() ?? Loc[nameof(Resources.WaitingHealthDataStatusMessage)])"
+                                   EnableHighlighting="@(!string.IsNullOrEmpty(_filter))"
+                                   HighlightText="@_filter"
+                                   TextVisualizerTitle="@context.Name">
+                            <ContentBeforeValue>
+                                @if (context.HealthStatus is null)
+                                {
+                                    <FluentProgressRing Width="16px" Style="display: inline-block;" Class="severity-icon" />
```

Comment:
FluentProgressRing is displayed like an icon here, and so should use the same styling, but with the added complication of it needing a different display value. I'm of the opinion that it could be bug-prone to extract it to a separate class if we change severity-icon behavior in the future